docs/connect.html

```diff
@@ -543,6 +543,67 @@ <h6>ACL requirements</h6>
+
+    <p>Prior to version 3.6, this strategy was not configurable, and behaved like the <code>ONLY_SCAN</code> mode which is compatible with all plugins. For version 3.6 and later, this mode defaults to <code>HYBRID_WARN</code> which is also compatible with all plugins, but logs a warning for all plugins which are incompatible with the other modes. For unit-test environments that use the <code>EmbeddedConnectCluster</code> this defaults to the <code>HYBRID_FAIL</code> strategy, which stops the worker with an error if an incompatible plugin is detected. Finally, the <code>SERVICE_LOAD</code> strategy will silently hide incompatible plugins and make them unusable.</p>
+
+    <h5><a id="connect_plugindiscovery_compatibility" href="#connect_plugindiscovery_compatibility">Verifying Plugin Compatibility</a></h5>
+
+    <p>To verify if all of your plugins are compatible, first ensure that you are using version 3.6 or later of the Connect runtime. You can then perform one of the following checks:</p>
+
+    <ul>
+        <li>Start your worker with the default <code>HYBRID_WARN</code>strategy, and WARN logs enabled for the <code>org.apache.kafka.connect</code> package. At least one WARN log message mentioning the <code>plugin.discovery</code> configuration should be printed. This log message will explicitly say that all plugins are compatible, or list the incompatible plugins.</li>
+        <li>Start your worker in a test environment with <code>HYBRID_FAIL</code>. If all plugins are compatible, startup will succeed. If at least one plugin is not compatible the worker will fail to start up, and all incompatible plugins will be listed in the exception.</li>
+    </ul>
+
+    <p>If the verification step succeeds, then your current set of installed plugins are compatible, and it should be safe to change the <code>plugin.discovery</code> configuration to <code>SERVICE_LOAD</code>. If you change the set of already-installed plugins, they may no longer be compatible, and you should repeat the above verification. If the verification fails, you must address the incompatible plugins before using the <code>SERVICE_LOAD</code> strategy.</p>
+
+    <h5><a id="connect_plugindiscovery_migrateartifact" href="#connect_plugindiscovery_migrateartifact">Operators: Artifact Migration</a></h5>
+
+    <p>As an operator of Connect, if you discover incompatible plugins, there are multiple ways to try to resolve the incompatibility. They are listed below from most to least preferable.</p>
+
+    <ol>
+        <li>Upgrade your incompatible plugins to the latest release version from your plugin provider.</li>
```

Comment:
This is a little misleading; someone may stop reading here and assume that an upgrade is guaranteed to fix the problem.

Maybe "Check the latest release of your plugin and, if it is compatible, upgrade to that version"?

Reply:
> This is a little misleading; someone may stop reading here and assume that an upgrade is guaranteed to fix the problem.

I couch this in the previous paragraph: 

> there are multiple ways to _try_ to resolve the incompatibility

> Maybe "Check the latest release of your plugin and, if it is compatible, upgrade to that version"?

That "Check" is the verification steps listed above. I'll call out that after addressing the plugins, the verification should be re-run.

Reply:
My intention with the language "Check the latest release" is to imply searching documentation, release notes, even source code, before installing that new version onto your Connect worker, which takes more effort from cluster administrators and introduces the usual risks associated with bumping any dependency.